### PR TITLE
Adds ember-models-table as basic component for all tables

### LINKED ISSUE #80: Convert events table to use ember-models-table
We want all tables to supporting sorting, filtering, searching. And for that we'll be using [ember-models-table](https://github.com/onechiporenko/ember-models-table) 

The library by default supports bootstrap. But we can customise the classes used so it can be made to support semantic UI. 

Let's start this on the events table located at `/events/live`

### PR DESCRIPTION
**Pull request by geekyd**
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Adds ui-table by extending models-table from ember-models-table, with support for semantic ui library. Adds custom cell templates for custom styling of cells in the table. 

Fixes #80 
@niranjan94 Please review.

### CODE REVIEW

File: app/templates/components/ui-table/header-row-sorting.hbs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<tr>
+  {{#each processedColumns as |column|}}
+    {{#if column.isVisible}}
+      {{#if column.useSorting}}
+        <th class="{{classes.theadCell}}">
+          <a class="ui" href="#" {{action "sort" column}}>
```
Comment (niranjan94) [inline app/templates/components/ui-table/header-row-sorting.hbs]: Let this not be an anchor. Make it a div or span that has an attribute `role="button"`

**Comment by geekyd** [comment]
@niranjan94 Made the changes please review.

**Comment by niranjan94** [comment]
@geekyd the changes requested (regarding import) doesn't seem to be done yet. Can you please check ?

**Comment by geekyd** [comment]
@niranjan94 Sorry I missed that, I have made the changes regarding import. Please check.

File: app/components/events/events-table.js [hunk 2]
```diff
@@ -1,6 +1,5 @@
-import Ember from 'ember';
+import uiTable from 'open-event-frontend/components/ui-table-server';
```
Comment (niranjan94) [inline app/components/events/events-table.js]: Since it's a class, let's call it `UiTable` or `UITable`

File: app/components/ui-table-select.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import uiSelect from 'ember-models-table/components/models-select';
```
Comment (niranjan94) [inline app/components/ui-table-select.js]: Since it's a class, let's call it `UiSelect` or `UISelect`

File: app/components/events/events-table.js (hunk 2 above)
Comment (geekyd) [inline app/components/events/events-table.js]: @niranjan94 Done!

File: app/components/ui-table-select.js (hunk 3 above)
Comment (geekyd) [inline app/components/ui-table-select.js]: @niranjan94 Done!

**Comment by geekyd** [comment]
@niranjan94 Made the changes. Please check

File: app/components/ui-table.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  navFirst        : 'chevron left icon',
+  navPrev         : 'chevron left icon',
+  navNext         : 'chevron right icon',
+  navLast         : 'chevron right icon',
+  caret           : 'caret',
+  expandRow       : 'plus icon',
+  expandAllRows   : 'plus icon',
+  collapseRow     : 'minus icon',
+  collapseAllRows : 'minus icon',
+  selectAllRows   : 'glyphicon glyphicon-check',
```
Comment (niranjan94) [inline app/components/ui-table.js]: These icons haven't been changed ?

File: app/components/ui-table.js [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+  paginationWrapperDefault       : 'column four wide',
+  buttonDefault                  : 'ui basic button',
+  noDataCell                     : '',
+  collapseRow                    : 'collapse-row',
+  collapseAllRows                : 'collapse-all-rows',
+  expandRow                      : 'expand-row',
+  expandAllRows                  : 'expand-all-rows',
+  thead                          : '',
+  input                          : 'form-control',
+  clearFilterIcon                : 'glyphicon glyphicon-remove-sign form-control-feedback',
```
Comment (niranjan94) [inline app/components/ui-table.js]: These icons haven't been changed ?

File: app/components/ui-table.js [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+  paginationWrapperNumeric       : 'column four wide',
+  paginationWrapperDefault       : 'column four wide',
+  buttonDefault                  : 'ui basic button',
+  noDataCell                     : '',
+  collapseRow                    : 'collapse-row',
+  collapseAllRows                : 'collapse-all-rows',
+  expandRow                      : 'expand-row',
+  expandAllRows                  : 'expand-all-rows',
+  thead                          : '',
+  input                          : 'form-control',
```
Comment (niranjan94) [inline app/components/ui-table.js]: This is a bootstrap class

**Comment by geekyd** [comment]
@niranjan94 I have removed all existing bootstrap classes. Please review.

**Comment by geekyd** [comment]
@niranjan94 I have removed all existing bootstrap classes. Please review.

**Comment by niranjan94** [comment]
@geekyd can you manually rebase this on top of `development`. ? Looks like there are conflicts during rebase. So, github cannot do it on its own. And I don't want to squash and merge these commits either. Each one is distinct and does something different.

**Comment by geekyd** [comment]
Okay I will try to rebase it on top of development.

**Comment by geekyd** [comment]
@niranjan94 I have rebased the PR. Please check.

File: app/templates/events.hbs [hunk 7]
```diff
@@ -17,9 +17,7 @@
       </div>
     </div>
   </div>
-  <div class="ui segment">
```
Comment (niranjan94) [inline app/templates/events.hbs]: Please revert this change. Not part of this PR.

Comment (geekyd) [inline app/templates/events.hbs]: @niranjan94 Done.

**Comment by niranjan94** [comment]
Deployment link: https://pr-102-open-event-frontend-fossasia.surge.sh

**Comment by niranjan94** [comment]
Deployment link: https://pr-102-open-event-frontend-fossasia.surge.sh
